docs/rfcs/0000-persistent-logging-component.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Persistent logging component
+layout: page
+permalink: /rfc/0000-persistent-logger
+---
+
+* Start date: 2020-10-10
+* RFC PR: [#0000](https://github.com/mozilla-mobile/android-components/pull/0000)
+
+## Summary
+Introduce a component for recording application logs on a device to a persistent store that can be shared with developers for debugging.
+
+## Motivation
+Reproducing bugs that a user encounters in the wild is a difficult process for the developer to get their device in a similar state in order to debug. We have useful information in our Android logs today that aid in debugging that we also do not have access to, but give us insight into the app state.
```

Comment:
Debug logging is quite useful for us, and but normally I would expect this to be stripped out from the APK.

Since we are limiting the scope of what is saved to within a specific time that is controlled by the user, it would give us useful granular information to have debug logging enabled. If it isn't at a performance or PII cost, we would revert the proguard the rules and control the logging flag to observe the recording state from this component instead.

Reply:
Just adding a note that A-C debug logs are actually just stripped out at run-time using a custom LogSink: https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/FenixLogSink.kt#L16

A-C logs can't be captured with ProGuard right now as we use `Log.println` (in A-Cs `AndroidLogSink`) which prevents filtering per log-level using ProGuard.

So, A-C debug logs are already available to this new component. We will have to decide how to handle GeckoView debug logs though (and other libs for that matter.) 

We could remove the ProGuard config and make sure they all go through the same log sink...

Reply:
Thanks! I misread the class/method name in the proguard file [here](https://github.com/mozilla-mobile/fenix/commit/0e2657e2620457c502e6b79b40f67ced6f8f0117#diff-24fc40bbf6fe503c74604e0477ff8f55aeab48cb521e1a844b9efa6c3771c15e) only affects `Log.d`.

Reply:
...a (configurable) LogSink in A-C that is, not Fenix.